### ISSUE TITLE
Perform Forge-style puppet syncs where content units already exist in Pulp

### ISSUE THREAD

**Issue by jeremycline** [PulpQE/pulp-smash#269]
This is from https://pulp.plan.io/issues/1937

The problem boiled down to the importer recognizing the content was already in Pulp, but failing to associate it with the repository.

The test steps would be as follows:

1. upload or sync a module to repo1
2. perform a sync on repo2 against a forge-style repository (one with a modules.json file) that includes the module from 1.
3. Assert the module from 1. is in the published repo2.

To do this in pulp-admin, it could be something as simple as:

1. pulp-admin puppet repo create --repo-id=repo1 --feed=http://forge.puppetlabs.com --queries=torssh
2. pulp-admin puppet repo sync run --repo-id=repo1
3. pulp-admin puppet repo create --repo-id=repo2 --feed=http://forge.puppetlabs.com --queries=torssh
4. pulp-admin puppet repo sync run --repo-id=repo2

**Comment by Ichimonji10** [comment]
Thanks for the clear explanation and the steps for reproducing. This looks easy to implement as an automated test case...

**Comment by danuzclaudes** [comment]
@Ichimonji10 Thank you for the comments!

While trying to extract the content-unit-counts for each repository, I found out a very subtle detail of this issue. The very first repository, at a clean state of db, would have counts as:
```
Content Unit Counts:  
  Puppet Module: 2
\n\n\n
```

However, for the rest of repositories, they would only have counts as:
```
Content Unit Counts: 
\n\n\n
```

And **even all those repositories are all deleted, the following new repos would still have no counts**, until an execution of `pclean`.

In that case. this pull request has been revised to react to the above output.

**Comment by Ichimonji10** [comment]
Fixed by #293.
